### PR TITLE
Fix: New order email for vendor is showing admin URL issue fixed #732

### LINKED ISSUE #732: New order email for vendor is showing admin URL
Description of the issue: 

The new order email has the admin order URL.

How to reproduce: 

- Purchase a product from a vendor 

- Check the email the vendor receives for a new order

- Click on the order number and the vendor will be redirected to wp-admin

Expected behavior:
 
A new order email for a vendor should redirect them to the vendor dashboard. 
-

### PR DESCRIPTION
**Pull request by alamgircsebd**
Need to first test it before merge.

### CODE REVIEW

File: includes/Emails/VendorNewOrder.php [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+ * @extends     WC_Email
+ */
+class VendorNewOrder extends WC_Email {
+
+    /**
+     * Constructor.
+     */
+    public function __construct() {
+        $this->id             = 'dokan_vendor_new_order';
+        $this->title          = __( 'Dokan Vendor New Order', 'woocommerce' );
```
Comment (saimonh3) [inline includes/Emails/VendorNewOrder.php]: Use `dokan-lite` text domain where necessary.

File: includes/Emails/VendorNewOrder.php [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+        return __( 'New Customer Order: #{order_number}', 'woocommerce' );
+    }
+
+    /**
+     * Trigger the sending of this email.
+     *
+     * @param int            $order_id The order ID.
+     * @param WC_Order|false $order Order object.
+     */
+    public function custom_trigger_email_action( $order_id, $order = false ) {
```
Comment (saimonh3) [inline includes/Emails/VendorNewOrder.php]: Rename the method to `trigger` to keep consistency.

File: includes/Emails/VendorNewOrder.php [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+            $order = wc_get_order( $order_id );
+        }
+
+        if ( is_a( $order, 'WC_Order' ) ) {
+            $this->object                           = $order;
+            $this->placeholders['{order_date}']     = wc_format_datetime( $this->object->get_date_created() );
+            $this->placeholders['{order_number}']   = $this->object->get_order_number();
+        }
+        
+        $sellers 	  = dokan_get_seller_id_by_order( $order_id );
```
Comment (saimonh3) [inline includes/Emails/VendorNewOrder.php]: Pay attention to indentation.

File: includes/Emails/VendorNewOrder.php [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+        if ( empty( $sellers ) ) {
+            return;
+        }
+        
+        // check has sub order 
+        if ( get_post_meta( $order_id, 'has_sub_order', true ) ) {
```
Comment (saimonh3) [inline includes/Emails/VendorNewOrder.php]: Can be used `dokan_is_sub_order` function to check whether the order is sub order or not.

File: templates/emails/vendor-new-order.php [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+ }
+
+$text_align = is_rtl() ? 'right' : 'left';
+
+ /**
+  * @hooked WC_Emails::email_header() Output the email header
+  */
+ do_action( 'woocommerce_email_header', $email_heading, $email );
+
+ $order_url = site_url().'/dashboard/orders/';
```
Comment (saimonh3) [inline templates/emails/vendor-new-order.php]: Don't hardcode the url. Use builtin dokan functions.

File: includes/Emails/VendorNewOrder.php [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+        
+        // check has sub order 
+        if ( dokan_is_sub_order( $order_id ) ) {
+        	foreach ($sellers as $seller) {
+        		$seller_info      = get_userdata( $seller );
+		        $seller_email 	  = $seller_info->user_email;
+		        $this->order_info = dokan_get_vendor_order_details( $order_id, $seller );
+			    $this->send( $seller_email, $this->get_subject(), $this->get_content(), $this->get_headers(), $this->get_attachments() );
+        	}
+        }else{
```
Comment (sabbir1991) [inline includes/Emails/VendorNewOrder.php]: Add proper spacing before and after else statement

File: templates/emails/vendor-new-order.php [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+ }
+
+$text_align = is_rtl() ? 'right' : 'left';
+
+ /**
+  * @hooked WC_Emails::email_header() Output the email header
+  */
+ do_action( 'woocommerce_email_header', $email_heading, $email );
+
+ $order_url = dokan_get_navigation_url( 'orders?order_id='.$order->get_order_number().'&_view_mode=email&permission=1' );
```
Comment (sabbir1991) [inline templates/emails/vendor-new-order.php]: Please use `add_query_args` function for rendering url and must use `esc_url` function for escaping

File: includes/functions.php [hunk 8]
```diff
@@ -2142,45 +2142,29 @@ function dokan_get_processing_time_value( $index ) {
...
+        if ( $vendor_id == $author_id ) {
+            $info['product']  = $item['name'];
+            $info['quantity'] = $item['quantity'];
+            $info['total']    = $item['total'];
+            array_push( $order_info, $info );
         }
     }
-
-    return apply_filters( 'dokan_email_recipient_new_order', $email );
+    return $order_info;
```
Comment (saimonh3) [inline includes/functions.php]: Maybe a filter is needed here.

File: includes/Emails/VendorNewOrder.php [hunk 9]
```diff
@@ -0,0 +1,194 @@
...
+            $this->placeholders['{order_number}'] = $this->object->get_order_number();
+        }
+        
+        $sellers = dokan_get_seller_id_by_order( $order_id );
+        if ( empty( $sellers ) ) {
+            return;
+        }
+        
+        // check has sub order 
+        if ( dokan_is_sub_order( $order_id ) ) {
```
Comment (saimonh3) [inline includes/Emails/VendorNewOrder.php]: If it's a sub-order then why we're looping through all the sellers? Shouldn't be when it's not a sub-order?

File: includes/functions.php (hunk 8 above)
Comment (alamgircsebd) [reply includes/functions.php]: on the function we just getting order details by order ID where no need any filter i think

File: includes/functions.php [hunk 10]
```diff
@@ -2142,45 +2142,29 @@ function dokan_get_processing_time_value( $index ) {
...
-                if ( $email != $seller_email ) {
-                    $email = $seller_email;
-                }
-            }
+function dokan_get_vendor_order_details( $order_id, $vendor_id ) {
+    $order      = wc_get_order( $order_id );
+    $info       = array();
+    $order_info = array();
+    foreach ( $order->get_items( 'line_item' ) as $item ) {
+        $product_id  = $item['product_id'];
```
Comment (saimonh3) [inline includes/functions.php]: @alamgircsebd What if it's a product variation? You are only fetching `product_id` from `line_items`.

File: includes/functions.php (hunk 8 above)
Comment (saimonh3) [reply includes/functions.php]: What if someone override the email template and want to show more information there?

File: includes/functions.php (hunk 10 above)
Comment (saimonh3) [reply includes/functions.php]: @sabbir1991 Please see.

Comment (saimonh3) [reply includes/functions.php]: @alamgircsebd  You may try this to get the product id from line items.

```php
$product_id = $item->get_product()->get_id();
```

File: includes/functions.php (hunk 8 above)
Comment (alamgircsebd) [reply includes/functions.php]: That line already remove which one used on old codes

**Pull request merged**